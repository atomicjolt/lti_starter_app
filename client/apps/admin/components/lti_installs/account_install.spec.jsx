import React from 'react';
import TestRenderer from 'react-test-renderer';
import AccountInstall from './account_install';

describe('lti installs account install', () => {

  let result;
  let instance;
  const accountInstalls = 123;
  const accountName = 'accountName';
  let clicked;

  const applicationInstance = {
    lti_key: 'lti_key'
  };
  const canvasRequest = () => { clicked = true; };

  describe('with account present', () => {
    clicked = false;
    beforeEach(() => {
      const account = {
        id: 12,
        name: accountName,
        external_tools: {
          consumer_key: 'consumer_key'
        }
      };

      result = TestRenderer.create(
        <AccountInstall
          applicationInstance={applicationInstance}
          canvasRequest={canvasRequest}
          accountInstalls={accountInstalls}
          account={account}
        />);
      instance = result.root;
    });

    it('renders', () => {
      expect(result).toBeDefined();
    });

    it('matches the snapshot', () => {
      expect(result).toMatchSnapshot();
    });

    it('handles the button click', () => {
      expect(clicked).toBeFalsy();
      instance.findByType('button').props.onClick();
      expect(clicked).toBeTruthy();
    });
  });

  describe('with account present', () => {
    beforeEach(() => {
      result = TestRenderer.create(
        <AccountInstall
          applicationInstance={applicationInstance}
          canvasRequest={canvasRequest}
          accountInstalls={accountInstalls}
        />);
      instance = result.root;
    });

    it('renders a header h3', () => {
      const h3 = instance.findByType('h3');
      expect(h3.props.children).toBe('Root');
    });

    it('renders buttons', () => {
      const accountButton = instance.findByType('button');
      expect(accountButton.props.children).toBe('Install Into Account');
    });
  });
});
