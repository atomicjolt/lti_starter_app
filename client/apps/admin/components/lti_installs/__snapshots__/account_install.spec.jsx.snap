// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lti installs account install with account present matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AccountInstall
    account={
      Object {
        "external_tools": Object {
          "consumer_key": "consumer_key",
        },
        "id": 12,
        "name": "accountName",
      }
    }
    accountInstalls={123}
    applicationInstance={
      Object {
        "lti_key": "lti_key",
      }
    }
    canvasRequest={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="c-title"
        >
          <h1>
            123
          </h1>
          <h3>
            accountName
          </h3>
        </div>,
        <button
          className="c-btn c-btn--yellow"
          onClick={[Function]}
        >
          Install Into Account
        </button>,
      ],
      "className": "c-info",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              123
            </h1>,
            <h3>
              accountName
            </h3>,
          ],
          "className": "c-title",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 123,
            },
            "ref": null,
            "rendered": 123,
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "accountName",
            },
            "ref": null,
            "rendered": "accountName",
            "type": "h3",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Install Into Account",
          "className": "c-btn c-btn--yellow",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Install Into Account",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="c-title"
          >
            <h1>
              123
            </h1>
            <h3>
              accountName
            </h3>
          </div>,
          <button
            className="c-btn c-btn--yellow"
            onClick={[Function]}
          >
            Install Into Account
          </button>,
        ],
        "className": "c-info",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                123
              </h1>,
              <h3>
                accountName
              </h3>,
            ],
            "className": "c-title",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 123,
              },
              "ref": null,
              "rendered": 123,
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "accountName",
              },
              "ref": null,
              "rendered": "accountName",
              "type": "h3",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Install Into Account",
            "className": "c-btn c-btn--yellow",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Install Into Account",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
